.github/workflows/pre-release.yml

```diff
@@ -16,7 +16,22 @@ env:
   NIMFLAGS: "--parallelBuild:${NPROC}"
 
 jobs:
+  tag-name:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Vars
+      id: vars
+      run: |
+        TAG=$([[ "${{github.ref}}" == "refs/heads/master" ]] && echo "${{env.RELEASE_NAME}}" || echo ${{github.ref}} | sed 's#refs/tags/##')
```

Comment:
We don't tend to have many frequent PRs being merged. However, I suggest creating a separate _release branch_, e.g. `release/0.20`, each time we create a release so that we make sure that nothing else gets added inadvertently during the release process. (A possible improvement for the future.)

Reply:
Yeah, we do that already - the check for `master` here is because the same PR is used for `nightly`, so if the build comes from `master`, rather than `v*-rc.*`, we want to call it `nightly`